### PR TITLE
TLS configure for OIDC connector

### PR DESCRIPTION
**Pull request by xtremerui**
This PR adds two configurable fiels `RootCAs` and `InsecureSkipVerify` to OIDC connector. When openning a connection to OIDC connect provider, it will use a httpclient based on these two fields.

### CODE REVIEW

**Comment by xtremerui** [comment]
@sagikazarmark just a quick raincheck here to see if this PR falls in the path. I am seeing some connectors providing TLS config (more or less). For example,

Github:
https://github.com/dexidp/dex/blob/6cdbb594064740794a79f30da83be92ec06c22e8/connector/github/github.go#L99-L108

LDAP:
https://github.com/dexidp/dex/blob/6cdbb594064740794a79f30da83be92ec06c22e8/connector/ldap/ldap.go#L255-L269

openshift:
https://github.com/dexidp/dex/blob/6cdbb594064740794a79f30da83be92ec06c22e8/connector/openshift/openshift.go#L81-L84

Thank you for your time!

**Comment by xtremerui** [comment]
Hi could I get some feedbacks on thie PR? Thx @sagikazarmark

**Review by sagikazarmark — CHANGES_REQUESTED** [review]
Apologies for the delay on this one.

I had a couple notes, but looks OK overall.

File: connector/oidc/oidc.go [hunk 1]
```diff
@@ -166,6 +188,40 @@ func (c *Config) Open(id string, logger log.Logger) (conn connector.Connector, e
 	}, nil
 }
 
+func newHTTPClient(rootCAs []string, insecureSkipVerify bool) (*http.Client, error) {
```
[the remark this answers is not in the document]
Comment (xtremerui) [reply connector/oidc/oidc.go]: I have extract the func out to a package named `httpclient`. The test of it requires some test certs as you can see. The instruction for regenerate the test certs are in the readme. Test certs are set to expired in 10 years.

Please let me know if you want set up the test differently. Thank you!

**Comment by nabokihms** [comment]
@xtremerui Thanks for your work, and sorry for this PR being delayed for so long. My apologies.

I am ready to merge this. Could you please fix the conflicts?

**Comment by xtremerui** [comment]
@nabokihms thx its done.

**Review by nabokihms — APPROVED** [review]
LGTM

**Pull request merged**